### PR TITLE
Wire up GetDetailsCall

### PR DESCRIPTION
**Pull request by PEConn**
This PR creates the DigitalGoodsRequestHandler which can accept and respond to calls for `getDetails`.

### CODE REVIEW

File: playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/DigitalGoodsRequestHandler.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+import java.util.Arrays;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.browser.trusted.TrustedWebActivityCallbackRemote;
+
+public class DigitalGoodsRequestHandler implements ExtraCommandHandler {
+    private final BillingWrapper mWrapper;
+    private final BillingWrapper.Listener mListener = new BillingWrapper.Listener() {
```
Comment (padr31) [inline playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/DigitalGoodsRequestHandler.java:20]: The mListener seems unused.

File: playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/DigitalGoodsRequestHandler.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    @NonNull
+    @Override
+    public Bundle handleExtraCommand(Context context, String commandName, Bundle args,
+            @Nullable TrustedWebActivityCallbackRemote callback) {
+        DigitalGoodsCallback wrappedCallback = (callbackName, callbackArgs) -> {
+            try {
+                callback.runExtraCallback(callbackName, callbackArgs);
+            } catch (RemoteException e) {
+                // The remote app crashed/got shut down, not much we can do.
```
Comment (padr31) [inline playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/DigitalGoodsRequestHandler.java:47]: Should we include a LOG message here?

File: playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/DigitalGoodsRequestHandler.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                if (getDetailsCall == null) break;
+
+                mWrapper.querySkuDetails(getDetailsCall.itemIds, getDetailsCall::respond);
+
+                return true;
+            case AcknowledgeCall.COMMAND_NAME:
+                AcknowledgeCall ackCall = AcknowledgeCall.create(args, callback);
+                if (ackCall == null) break;
+
+                // TODO(peconn): Implement
```
Comment (padr31) [inline playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/DigitalGoodsRequestHandler.java:72]: NIT: I am not very comfortable with TODOs left in code. If you plan to implement it soon, would it be better to implement it in this PR? Or make the PR work in progress?

File: playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/DigitalGoodsRequestHandler.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                return true;
+        }
+        return false;
+    }
+}
```
Comment (padr31) [inline playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/DigitalGoodsRequestHandler.java:77]: NIT: Missing new line, possibly elsewhere as well

File: playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/DigitalGoodsRequestHandler.java (hunk 3 above)
Comment (padr31) [reply playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/DigitalGoodsRequestHandler.java:72]: Or maybe it's enough to give a comment on what implementation is still missing and why

File: playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/DigitalGoodsRequestHandler.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
+package com.google.androidbrowserhelper.playbilling.digitalgoods;
```
Comment (andreban) [inline playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/DigitalGoodsRequestHandler.java:1]: reminder: license headers.

File: playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/DigitalGoodsRequestHandler.java (hunk 1 above)
Comment (andreban) [reply playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/DigitalGoodsRequestHandler.java:20]: It's passed as parameter to `BillingWrapperFactory.get()` later - do you mean we don't need it as an instance variable maybe?

**Review by andreban — APPROVED** [review]
LGTM, with added license headers and missing EOL.

File: playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/DigitalGoodsRequestHandler.java (hunk 1 above)
Comment (padr31) [reply playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/DigitalGoodsRequestHandler.java:20]: I meant that it's methods were empty and then it seemed redundant.

**Review by padr31 — APPROVED** [review]

File: playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/DigitalGoodsRequestHandler.java (hunk 1 above)
Comment (padr31) [reply playbilling/src/main/java/com/google/androidbrowserhelper/playbilling/digitalgoods/DigitalGoodsRequestHandler.java:20]: But I see that it's filled in in the other PR https://github.com/GoogleChrome/android-browser-helper/pull/163/files
so it's ok.

**Pull request merged**